### PR TITLE
Handle Publisher.isAbandoned + integrity checks.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: app/lib/publisher/models.dart [hunk 1]
```diff
@@ -46,10 +40,8 @@ class Publisher extends db.ExpandoModel {
   /// ensure that we can see:
   /// (A) who uploaded a package, and,
   /// (B) who granted the permissions that allowed said package to be uploaded.
-  bool get isAbandoned => isAbandoned == true;
-  set isAbandoned(bool value) {
-    isAbandonedFlag = value;
-  }
+  @db.BoolProperty(required: true)
+  bool isAbandoned;
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/publisher/models.dart]: It is for packages that were uploaded under a publisher, which got later abandoned. We should probably make it clear on the UI that the publisher is abandoned, and best to do with a flag, as derived flag may not be future-proof.

File: app/lib/shared/integrity.dart [hunk 2]
```diff
@@ -113,6 +126,42 @@ class IntegrityChecker {
...
+      if (p.isAbandoned) {
+        _publishersAbandoned.add(p.publisherId);
+        if (members.isNotEmpty) {
+          _problems.add('Publisher(${p.publisherId}) is marked as abandoned, '
+              'but has members (first: ${members.first.userId}).');
+        }
+        if (members.isEmpty && p.contactEmail != null) {
+          _problems.add(
+              'Publisher(${p.publisherId}) is marked as abandoned, has no members, '
+              'but still has contact email (${p.contactEmail}).');
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/shared/integrity.dart]: Why would we display a contact e-mail once the publisher has no admin? They have decided to leave it behind...

File: app/lib/publisher/models.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonasfj) [reply app/lib/publisher/models.dart]: oh, nvm, I read it wrong, yes agree... my comment made no sense... I was talking about the stuff you removed :)

File: app/lib/shared/integrity.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonasfj) [reply app/lib/shared/integrity.dart]: If the contact email was setup to be a mailing list, then this seems fine...

We're not currently supporting this... but we probably will in the future allow users to set any contact email, so long as they can verify it.

[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/shared/integrity.dart]: We'll remove it at that point :)

**Pull request merged**